sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -357,29 +358,47 @@ def _initial_request(self):
         )
 
         try:
-            location_mode, response = self._clients.blob.download(
-                range=range_header,
-                range_get_content_md5=range_validation,
-                validate_content=self._validate_content,
-                data_stream_total=None,
-                download_stream_current=0,
-                **self._request_options
-            )
 
-            # Check the location we read from to ensure we use the same one
-            # for subsequent requests.
-            self._location_mode = location_mode
+            retry_active = True
+            retry_total = 3
+            while retry_active:
+                location_mode, response = self._clients.blob.download(
+                    range=range_header,
+                    range_get_content_md5=range_validation,
+                    validate_content=self._validate_content,
+                    data_stream_total=None,
+                    download_stream_current=0,
+                    **self._request_options
+                )
 
-            # Parse the total file size and adjust the download size if ranges
-            # were specified
-            self._file_size = parse_length_from_content_range(response.properties.content_range)
-            if self._end_range is not None:
-                # Use the end range index unless it is over the end of the file
-                self.size = min(self._file_size, self._end_range - self._start_range + 1)
-            elif self._start_range is not None:
-                self.size = self._file_size - self._start_range
-            else:
-                self.size = self._file_size
+                # Check the location we read from to ensure we use the same one
+                # for subsequent requests.
+                self._location_mode = location_mode
+
+                # Parse the total file size and adjust the download size if ranges
+                # were specified
+                self._file_size = parse_length_from_content_range(response.properties.content_range)
+                if self._end_range is not None:
+                    # Use the end range index unless it is over the end of the file
+                    self.size = min(self._file_size, self._end_range - self._start_range + 1)
+                elif self._start_range is not None:
+                    self.size = self._file_size - self._start_range
+                else:
+                    self.size = self._file_size
+
+                try:
+                    self._current_content = process_content(
+                        response,
+                        self._initial_offset[0],
+                        self._initial_offset[1],
+                        self._encryption_options
+                    )
+                    retry_active = False
+                except (requests.exceptions.ChunkedEncodingError, requests.exceptions.ConnectionError) as error:
```

Comment:
For example if you get a partial gzip file and try to unzip you will get ContentDecodingError, we want to throw immediately since retry wouldn't help.